app/decorators/user_decorator.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class UserDecorator < Draper::Decorator
+  delegate_all
+
+  def greeting
+    if first_name? && last_name?
```

Comment:
better to use guard here :)

```
def greeting 
  I18n.t('user.greeting', user: email) if first_name.nil? && last_name.nil? 
  # I18n.t('user.greeting', user: email) if full_name? # def fullname;  first_name.nil? && last_name.nil?; end;
  I18n.t('user.greeting', user: "#{first_name} #{last_name}")
end
```

in this case you can also do smth like 

```
def greeting 
  user = fullname? ? "#{first_name} #{last_name}" : email 
  
  I18n.t('user.greeting', user: user)
end
```

i'm not sure also here about naming `user` can be misleading because i'm expecting the user model here - wdyt about smth like `user_welcome_label`